src/bundle/Controller/ContentController.php

```diff
@@ -381,4 +383,78 @@ public function updateMainTranslationAction(Request $request): Response
 
         return $this->redirectToRoute('ezplatform.dashboard');
     }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function hideAction(Request $request): Response
+    {
+        $form = $this->createForm(ContentType::class);
+        $form->handleRequest($request);
+
+        if (!$form->isSubmitted() || !$form->isValid()) {
+            throw new BadRequestHttpException();
+        }
+
+        /** @var \EzSystems\EzPlatformAdminUi\Form\Data\Content\ContentData $contentData */
+        $contentData = $form->getData();
+        $contentInfo = $contentData->getContentInfo();
+        $location = $contentData->getLocation();
+
+        if ($contentInfo->isHidden) {
+            throw new BadRequestHttpException();
+        }
+
+        $this->contentService->hideContent($contentInfo);
+
+        $this->notificationHandler->success(
+            $this->translator->trans(
+                /** @Desc("Content '%name%' has been hidden.") */
+                'content.hide.success',
+                ['%name%' => $contentInfo->name],
+                'content'
+            )
+        );
+
+        return $this->redirectToLocation($location);
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function revealAction(Request $request): Response
+    {
+        $form = $this->createForm(ContentType::class);
+        $form->handleRequest($request);
+
+        if (!$form->isSubmitted() || !$form->isValid()) {
+            throw new BadRequestHttpException();
+        }
+
+        /** @var \EzSystems\EzPlatformAdminUi\Form\Data\Content\ContentData $contentData */
+        $contentData = $form->getData();
+        $contentInfo = $contentData->getContentInfo();
+        $location = $contentData->getLocation();
```

Comment:
in `EzSystems\EzPlatformAdminUi\Form\Data\Content\ContentData` getLocation() can return `null` and below we have `redirectToLocation()`. Will not it be a problem?